### ISSUE TITLE
[029]: unable to find child 'IsTruncated':0 in node 'ListAllMyBucketsResult'

### ISSUE THREAD

**Issue by cbid71** [minio/minio#19274]
## NOTE
If this case is urgent, please subscribe to [Subnet](https://min.io/pricing) so that our 24/7 support team may help you faster.

I'm using pgbackrest to export logs for a postgresql database to a MinIO bucket but I can't create a stanza ( kind of configuration bound) because of what seems to be an incompatibility of API. 

I've took a look with pgbackrest github support and I've seen this subject : [https://github.com/pgbackrest/pgbackrest/issues/1981](https://github.com/pgbackrest/pgbackrest/issues/1981) with a possible problem of support for new version of S3 bucket API. 

## Expected Behavior

A Minio bucket should contain files from an export. 

## Current Behavior

We have this problem on the creation of a stanza :
 
```
2024-03-15 16:38:46.338 P00  DEBUG:     command/exit::exitSafe: (result: 0, error: true, signalType: 0)
ERROR: [029]: unable to find child 'IsTruncated':0 in node 'ListAllMyBucketsResult'
```

pgbackrest jobs can't connect correctly to the bucket.
According to the pgbackrest support it should be related to a problem of API versions between v2 and v1 : 

https://github.com/pgbackrest/pgbackrest/issues/1827
https://github.com/pgbackrest/pgbackrest/issues/1981
https://github.com/pgbackrest/pgbackrest/issues/2101

I quote here : 

```
IsTruncated is a required field for ListObjectsV2 (https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjectsV2.html), which is what we request from the server. Unfortunately some servers ignore this request and return V1 results, which have been deprecated for eight years or more.

AWS CLI fails back to V1, we don't. Contact your vendor about getting this fixed (we've seen other vendors fix it when the issue is pointed out to them).

You can search issues for IsTruncated to get more info on past similar issues.
```

## Steps to Reproduce (for bugs)

1. here a pgbackrest configuration 

```
[global]
log-level-console=warn
repo1-path=/saves
repo1-retention-diff=3
repo1-retention-full=2
repo1-retention-full-type=count
repo1-s3-bucket=mybucket
repo1-s3-endpoint=http://192.168.10.10
repo1-s3-host=http://192.168.10.10
repo1-s3-port=9000
repo1-s3-verify-tls=n
repo1-s3-key=xxxxxxxxx
repo1-s3-key-secret=xxxxxxxx
repo1-s3-region=us-east-1
repo1-type=s3

start-fast=y

```

3. here is a command

`
pgbackrest --stanza=test --pg1-path=/data stanza-create
`

## Context

I'd like to export files from a a postgresql database savegard to a MinIO bucket.

## Regression

Is this issue a regression? According to the pgbackrest team it work up to the 2.39, with 2.40 they have removed the support of APIv1 for buckets, leading to the problem.
 

## Your Environment

pgbackrest : 2.48
MinIO : 2024-02-24T17-11-14Z

**Comment by harshavardhana** [comment]
@cbid71 please capture `mc admin trace -v` output when you run the pgrestbackup

**Comment by harshavardhana** [comment]
> 2024-03-15 16:38:46.338 P00  DEBUG:     command/exit::exitSafe: (result: 0, error: true, signalType: 0)
> ERROR: [029]: unable to find child 'IsTruncated':0 in node 'ListAllMyBucketsResult'

This looks like ListBuckets() call was made not ListObjects() - upstream didn't even debug this properly what a shame https://github.com/pgbackrest/pgbackrest/issues/2101

Instead of doing ListObjects() path style they are performing ListObjects() - bucket DNS style addressing. 

You need to run 

--repo-s3-uri-style path

With MinIO by default if you haven't configured MinIO to support Bucket DNS addressing.

**Comment by cbid71** [comment]
Hello,
I'm so sorry for the delay in the answer I was away.
Thank you, you are right in the end I fixed the problem by using the additional 

`repo1-s3-uri-style=path`

in my config file and it worked.

Another way of doing it : 

`pgbackrest --stanza=test --pg1-path=/data stanza-create --repo1-s3-uri-style=path`

I tried it before but it didn't work, that's what motivated me to open this issue, I probably broke something else while turning my configuration hundreds of times to make it work.

Thank you
